### PR TITLE
User sign in page

### PR DESCRIPTION
**Pull request by jubuyer**
Changes:
- Adds `login.jsx` component for user login
- Updates `header.jsx` component with sign in page
- Updates `./client/Dockerfile` optimization for copying packages

To Test:
- Follow [src](https://github.com/csci-499-sp24/FaceStylizer/blob/feature/LoginPage/README.md) `README` instructions to build docker container and view webpage at http://localhost:3000/

### CODE REVIEW

**Comment by jubuyer** [comment]
image

Current UI of sign in page. Connection to our database for account fetching to be implemented next.

File: client/components/header.jsx
Comment (George-Sucuzhanay) [inline client/components/header.jsx:1]: thank you for adding the comment on where to add additional tags

File: client/components/login.jsx
Comment (George-Sucuzhanay) [inline client/components/login.jsx:1]: great use of reactstrap library

**Review by George-Sucuzhanay — APPROVED** [review]

File: client/components/header.jsx [hunk 1]
```diff
@@ -1,29 +1,32 @@
...
+  const [open, setOpen] = useState(true);
+
+  return (
+    <div className="antialiased bg-white dark-mode:bg-gray-800">
+      <div className="w-full text-gray-700 dark-mode:text-gray-200">
+        <div className="flex flex-col max-w-screen-xl px-4 mx-auto md:items-center md:justify-between md:flex-row md:px-6 lg:px-8">
+          <div className="flex flex-row items-center justify-between p-4">
+            <a href="#" className="text-lg font-semibold tracking-widest text-gray-900 uppercase rounded-lg dark-mode:text-white focus:outline-none focus:shadow-outline">FACESTYLIZER</a>
+            {/* <button className="rounded-lg md:hidden focus:outline-none focus:shadow-outline" onClick={() => setOpen(!open)}>
+              <svg fill="currentColor" viewBox="0 0 20 20" className="w-6 h-6">
```
Comment (EricLi-Dev) [inline client/components/header.jsx:16]: Can we remove these comments if they're not needed?

File: client/components/login.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            Sign In
+          </h5>
+        </div>
+        <ModalBody>
+          <Form>
+            <FormGroup>
+              <Label for="exampleEmail">
+                Email
+              </Label>
+              {/* <InputGroup className=" input-group-alternative">
```
Comment (EricLi-Dev) [inline client/components/login.jsx:41]: .

**Review by EricLi-Dev — CHANGES_REQUESTED** [review]
If you can remove those comments we don't use.

File: client/components/header.jsx (hunk 1 above)
Comment (jubuyer) [reply client/components/header.jsx]: Resolved.

File: client/components/login.jsx (hunk 2 above)
Comment (jubuyer) [reply client/components/login.jsx]: Resolved.

File: client/components/header.jsx [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
+
+  return (
+    <div className="antialiased bg-white dark-mode:bg-gray-800">
+      <div className="w-full text-gray-700 dark-mode:text-gray-200">
+        <div className="flex flex-col max-w-screen-xl px-4 mx-auto md:items-center md:justify-between md:flex-row md:px-6 lg:px-8">
+          <div className="flex flex-row items-center justify-between p-4">
+            <a href="#" className="text-lg font-semibold tracking-widest text-gray-900 uppercase rounded-lg dark-mode:text-white focus:outline-none focus:shadow-outline">FACESTYLIZER</a>
+          </div>
+          <nav className={`flex-col flex-grow ${open ? '' : 'hidden'} pb-4 md:pb-0 md:flex md:justify-end md:flex-row`}>
+            {/* <a className="px-4 py-2 mt-2 text-sm font-semibold bg-transparent rounded-lg dark-mode:bg-transparent dark-mode:hover:bg-gray-600 dark-mode:focus:bg-gray-600 dark-mode:focus:text-black dark-mode:hover:text-black dark-mode:text-black md:mt-0 md:ml-4 hover:text-black focus:text-black hover:bg-gray-200 focus:bg-gray-200 focus:shadow-outline"> */}
```
Comment (EricLi-Dev) [inline client/components/header.jsx:17]: .

**Review by EricLi-Dev — CHANGES_REQUESTED** [review]
last one

File: client/components/header.jsx (hunk 3 above)
Comment (jubuyer) [reply client/components/header.jsx:17]: Sorry

**Review by EricLi-Dev — APPROVED** [review]
lgtm

**Pull request merged**